doc/search-api.md

```diff
@@ -46,176 +59,190 @@ The parameters supported are:
+   filters.  If set to `exclude_field_filter`, the aggregate counts are made
+   after applying all filters _except_ for those applied to the field that
+   the aggregates are being counted for.  This is a convenient option for
+   calculating values to show in common interfaces which use aggregate for
+   narrowing down search results.
+
+ - `order`: Colon separated list of ordering types.
+
+   The available ordering types are:
+
+    - `filtered`: whether the value is used in an active filter.  This can be
+used to sort such that the values which are being filtered on come
+first.
+    - `count`: order by the number of documents in the search matching the
+facet value.
+    - `value`: sort by value if the field values are string, sort by the
+`title` field in the value object if the value is an object.  Sorting
+is case insensitive in either case.
+    - `value.slug`: the slug in the facet value object
+    - `value.link`: the link in the facet value object
+    - `value.title`: the title in the facet value object (case insensitive)
+
+   Each ordering may be preceded by a "-" to sort in descending order.
+   Multiple orderings can be specified, in priority order, separated by a
+   colon.  The default ordering is "filtered:-count:slug".
+
+ - `examples`: integer number of example values to return
+
+   This causes facet values to contain an "examples" hash as an additional
+   field, which contains details of example documents which match the query.
+   The examples are sorted by decreasing popularity.  An example facet value
+   in a response with this option set as "examples:1" might look like:
+
+      "value" => {
+        "slug" => "an-example-facet-slug",
+        "example_info" => {
+          "total" => 3,  # The total number of matching examples
+          "examples" => [
+            {"title" => "Title of the first example", "link" => "/foo"},
+          ],
+        }
+      }
+
+ - `example_scope`: `global` or `query`.  If the `examples` option is supplied, the
+   `example_scope` option must be supplied too.
+
+   The value of `global` causes the returned examples to be taken from all
+   documents in which the facet field has the given slug.
+
+   The value of `query` causes the returned examples to be taken only from
+   those documents which match the query (and all filters).
+
+ - `example_fields`: colon separated list of fields.
+
+   If the examples option is supplied, this lists the fields which are
+   returned for each example.  By default, only a small number of fields are
+   returned for each.
+
+## Other parameters
+- ab_tests: a/b test with selected variant type. This allows test to be configured
+  from upstream apps.
+
+  Each a/b test name should be followed by a ':' and then the variant type to
+  be used. If multiple a/b test setting are being passed in they should be
+  comma separated.
+
+  No validation is done to ensure the a/b test name provided is current implemented.
```

Comment:
Do we want to mention cache busting?

Reply:
Have added a short description at the end. There's probably a couple of other params missing but I'm just focusing on the main ones for now.